### PR TITLE
Video playlist

### PR DESCRIPTION
**Pull request by jeetbhalani15**
## Playlist Management for videos

-  Added playlist Page
-  Added playlist features => ( add to playlist , remove playlist , remove playlist videos ) using API 
-  Added Modal to create Playlist on home page & video page
-  Add & remove the playlist from the modal.

IGNORE:  Comments if any and hardcoded colors soon change them all 
 & will make seprate files for all services & api calls.

### CODE REVIEW

File: Gravity-tube/src/Components/Modal/Modal.css [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  width: 100%;
+  z-index: 15;
+  top: 0%; 
+  left: -1%;
+  background-color: #000000cc;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  }
+  .playlist-box{
```
Comment (Jay-Gadhiya) [inline Gravity-tube/src/Components/Modal/Modal.css:15]: - suggestion: give space between two classes, and also between curly brackets and class name.

File: Gravity-tube/src/Components/Modal/Modal.css [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      width: 18rem;
+      border: none;
+      background: transparent;
+      border: 2px solid #037c71;
+      border-radius: 5px;
+  }
+  .playlist-btn{
+    margin-top: 1rem;
+    padding: 0.5rem 6.5rem;
+    background-color: #009688;
```
Comment (Jay-Gadhiya) [inline Gravity-tube/src/Components/Modal/Modal.css:49]: -Try to use variable name

File: Gravity-tube/src/Components/Modal/Modal.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            value={playlistName.title}
+          />
+        </div>
+     { playlists?.map(({title:{title}, videos, _id: playlistId, isChecked})=>{
+       
+       return (
+        <div key={playlistId} className="label-box">
+        <label htmlFor={playlistId}>
+            <input type="checkbox" id={playlistId} 
+            checked ={videos?.find((item=> item._id === videoId ? true : false))}
```
Comment (Jay-Gadhiya) [inline Gravity-tube/src/Components/Modal/Modal.js:72]: -you can put  `{videos?.find((item=> item._id === videoId)) ? true : false}`, so checked get boolean values and not undefined.

File: Gravity-tube/src/Components/Video-Listing/VideoListing.js [hunk 4]
```diff
@@ -1,20 +1,16 @@
...
     <div className="video-box">
       <h2>Recommended</h2>
       <div className="videos-container">
-        {videoDataState.videos.map((video)=>(
-           <VideoCard 
-           livenow = {true}
-           video={video}
-         />
+        {videoDataState.videos.map((video) => (
+          <VideoCard livenow={true} video={video} />
```
Comment (Jay-Gadhiya) [inline Gravity-tube/src/Components/Video-Listing/VideoListing.js:13]: - suggestion: you can give prop name in camel case, `liveNow`

File: Gravity-tube/src/Reducers/PlaylistReducer.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+              item._id === payload._id
+                ? { ...item, videos: payload.videos }
+                : item
+            ),
+          };
+      case "REMOVE_PLAYLIST":
+          return  {
+            ...state,
+            playlists: payload
+          };
```
Comment (Jay-Gadhiya) [inline Gravity-tube/src/Reducers/PlaylistReducer.js:20]: - you can use your first case `SET_NEW_PLAYLIST` case here also instead of make new case

**Review by Jay-Gadhiya — APPROVED** [review]

File: Gravity-tube/src/Components/Modal/Modal.css [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  .playlist-text{
+    display: flex;
+    flex-direction: column;
+  }
+  .label-box{
+    display: grid;
+    grid-template-columns: 134px 127px;
+  }
+  .label-box > label{
+    color: #008387;
```
Comment (chanchal16) [inline Gravity-tube/src/Components/Modal/Modal.css:69]: Use variables for colors as some of the colors are repeated.

File: Gravity-tube/src/Components/Modal/Modal.js [hunk 7]
```diff
@@ -0,0 +1,123 @@
+
+import axios from "axios";
+import { Axios } from "axios";
+import React, { useState } from "react";
+import { FaSearch } from "react-icons/fa";
+import { IoClose } from "react-icons/io5";
+import { useAuth } from "../../Contexts/Auth-context";
+import { usePlaylist } from "../../Contexts/Playlist-context";
+import { useVideoData } from "../../Contexts/Videos-context";
```
Comment (chanchal16) [inline Gravity-tube/src/Components/Modal/Modal.js:10]: You can use barrel exports by exporting all context files in one main file.

File: Gravity-tube/src/Components/Modal/Modal.js [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+            {title}
+          </label>
+          </div>
+          )
+        
+     }) 
+     
+    }
+        
+
```
Comment (chanchal16) [inline Gravity-tube/src/Components/Modal/Modal.js:83]: remove the blank spaces

File: Gravity-tube/src/Pages/WatchLater/Components/WatchlaterVideos.css [hunk 9]
```diff
@@ -118,6 +118,62 @@ hr{
...
+    align-items: flex-start;
+}
+.playlist-info{
+    position: relative;
+    display: flex;
+    align-items: center;
+    grid-gap: 2rem;
+    gap: 11.8rem;
+}
+/* .playlist-card-area:hover{
```
Comment (chanchal16) [inline Gravity-tube/src/Pages/WatchLater/Components/WatchlaterVideos.css:136]: remove the commented code

File: Gravity-tube/src/Components/VideoCard/VideoCard.js [hunk 10]
```diff
@@ -55,8 +66,10 @@ function VideoCard({ video }) {
             </div>
           </div>
 
+
+
           {showOptions && (
-            <div ref={domNode} className="more-option-box">
+            <div ref={domRef} className="more-option-box">
               <div className="options">
               {videoDataState.watchLater.find((item)=>item._id === video._id) ? (
```
Comment (chanchal16) [inline Gravity-tube/src/Components/VideoCard/VideoCard.js:74]: As this line of code is repeating in multiple files you can create a utility function for it. This way you won't have to write it  multiple times.

**Pull request merged**